extmod/modnetwork.h

```diff
@@ -82,15 +77,18 @@ typedef struct _mod_network_socket_obj_t {
             uint8_t domain;
             uint8_t type;
             int8_t fileno;
+            uint8_t bound;
         } u_param;
         mp_uint_t u_state;
     };
+    uint16_t timeout;
```

Comment:
did you intend to keep this timeout entry here?  I guess it can be removed altogether, or at least put in `MICROPY_PY_USOCKET_EXTENDED_STATE` below?

Reply:
Yes, I was just commenting about it below, I realized it can also be used for `blocking` for unix sockets (timeout == 0 blocking) and I already use it in the nina and winc driver, so might be a good idea to keep it ? Otherwise, maybe make it unconditionally as a union with `void *state` ?

Reply:
Note if it's removed, it will have to be allocated dynamically per socket for NICs and cast each time it's accessed, if we must remove it then I suggest using a union with state.

Reply:
Because a GC block is 4 words adding this timeout takes the size of the struct from 4 words to 5 words.  It does seem useful having `timeout` so maybe just also include `void *state` unconditionally because it won't use any extra RAM (the socket object will already be 2x GC blocks).

Reply:
@dpgeorge I can make them both configurable with EXTENDED_STATE if you like ? For ports that don't use timeout or need additional socket state, it will be 1 GC block.

Reply:
Yes, sounds good!

Reply:
Okay done.